src/core/obj.js

```diff
@@ -748,12 +760,16 @@ var Catalog = (function CatalogClosure() {
           }
           /* falls through */
         default:
-          warn('Catalog_parseDestDictionary: Unrecognized link type "' +
-               linkType + '".');
+          warn(`parseDestDictionary: Unsupported Action type "${actionName}".`);
           break;
       }
-    } else if (destDict.has('Dest')) { // Simple destination link.
-      dest = destDict.get('Dest');
+    } else if (destDict.has('Dest')) { // Simple destination.
+      let simpleDest = destDict.get('Dest');
+      if (isDict(simpleDest)) {
```

Comment:
How about directly assigning to `dest` instead of a temporary variable? In case `simpleDest` is invalid, you are already returning early, so not using a temporary variable should be fine.

Reply:
This was done because the (first version of the) patch only checked if an `A` entry existed, and not that it was actually a dictionary when doing the "is there a `Dest` entry and if so is it a dictionary" checks above. Hence it possible to end up here with a simple destination containing a dictionary.

I've updated the patch to do the proper checks above, so that these changes could be reverted.

Reply:
I updated the patch to do the proper checks above, so that we'll no longer reach this branch in the `destDict.get('Dest') instanceof Dict` case.